### PR TITLE
1.2.0 - documentation and hooks

### PR DESCRIPTION
**Pull request by adamdubiel**

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,177 @@
+# Makefile for Sphinx documentation
```
Comment (szpak) [inline docs/Makefile]: Wow, Makefile in Java project :).
Maybe there will  a Gradle plugin one day...

**Comment by martinda** [comment]
I have found an answer in the new proposed documentation that I could not find in the current doc (about Jenkins). And I really like the more extensive doc (Version Creation, the "Reading" section in particular). This reinforces the need for me to check that releases are only made from a single branch in a single repository (in my case master). I don't want users to do ad hoc releases from random branches just because they can type "gradle release". Anyway, I am not qualified to do a review of this new code, just giving you some comments about what I like.

File: docs/Makefile (hunk 1 above)
Comment (adamdubiel) [inline docs/Makefile]: I doubt, they would need gradle at readthedocs end :) This documentation is not build during axion-release build.

**Comment by adamdubiel** [comment]
@martinda that's exactly why i moved to readthedocs, readme could not contain all information i wanted to share anymore. Glad you like it :)
As to the "gradle release from random branches" - we use release only on CI, so it is easy to control it.

**Comment by martinda** [comment]
In the old documentation, there was a short section describing the main tasks of the plugin. In this description, the release task specifies that the tag is created and pushed to the remote. 
I could not find this information in the new doc.

**Comment by martinda** [comment]
The old documentation also had a long example, which I find useful when I need to see all the features at a glance. Can you please bring that back in the new doc? Thanks in advance.

**Comment by adamdubiel** [comment]
Okay, i will update the documentation before merge - thanks for the feedback and let me know if there is anything missing still.

**Comment by adamdubiel** [comment]
@martinda - i added missing bits of documentation

**Comment by martinda** [comment]
Thank you!

**Pull request merged**